app/services/notification/mailClient.scala

```diff
@@ -0,0 +1,48 @@
+package services.notification
+
+import javax.inject.Inject
+
+import com.amazonaws.handlers.AsyncHandler
+import com.amazonaws.services.simpleemail.AmazonSimpleEmailServiceAsyncClient
+import com.amazonaws.services.simpleemail.model._
+import com.google.inject.ImplementedBy
+import models.Email
+import play.api.{Configuration, Logger}
+import play.api.libs.concurrent.Execution.Implicits._
+
+import scala.concurrent.{Future, Promise}
+
+@ImplementedBy(classOf[AWSMailClient])
+trait MailClient {
```

Comment:
The idea (as mentioned also in a previous comment) is to abstract out the implementation library for sending the emails in case we would like to change it? 
Maybe this is causing noise though rather than helping, I don't know. :/